### PR TITLE
Improve searching

### PR DESCRIPTION
**Pull request by asmsuechan**
# context
This PR fixes a problem which IME doesn't work on searching.

# before
acbf4710df3def9ca0b72c7cc48bdb7e

# after
03d6e235a21cf0df2b7bf606e2caa934

93a92bff4cd6ff62e8d80186de2339c8

Boostnote will support IME input (I checked working in Japanese keyboard and Chinese pin yin keyboard)

### CODE REVIEW

File: browser/main/TopBar/index.js [hunk 1]
```diff
@@ -34,9 +37,52 @@ class TopBar extends React.Component {
...
 
+  handleKeyDown (e) {
+    // reset states
+    this.setState({
+      isAlphabet: false,
+      isIME: false
+    })
+
+    // When the key is an alphabet, del, enter or ctr
+    if (e.keyCode <= 90) {
```
Comment (asmsuechan) [inline browser/main/TopBar/index.js]: This does not work at Umlaut... Need to change.

Comment (asmsuechan) [reply browser/main/TopBar/index.js]: I decided to extend the keyCode scope to 186<=e.keyCode<=222 according to http://www.javascripter.net/faq/keycodes.htm.

**Comment by asmsuechan** [comment]
# current specifications
There are some unclear points.

* search by `#tag1 #tag2` returns the results of only `#tag2`
* search by `#tag1 content` is as same as AND search by `#tag1 content` (and #tag1 is regarded a word, not a tag)
* search by `content #tag1` returns the results of only `#tag1`

**Comment by lyrl** [comment]
chinese input has same problem

**Comment by asmsuechan** [comment]
@lyrl I know it. I have a question regarding Chinese keyboard. Is pinyin keyboard used by much Chinese?

> Boostnote will support IME input (I checked working in Japanese keyboard and Chinese pin yin keyboard)

**Comment by lyrl** [comment]
@asmsuechan SouGou pinyin is most popular in china! http://pinyin.sogou.com/

**Comment by asmsuechan** [comment]
@lyrl 谢谢你告诉我！I switched from Google pinyin keyboard to SouGou.

**Comment by KuangLei** [comment]
Btw, under Linux, most Chinese people will choose [fcitx](https://fcitx-im.org/wiki/Fcitx).

**Pull request merged**